cli/tests/test_git_colocated.rs

```diff
@@ -98,28 +98,59 @@ fn test_git_colocated() {
 fn test_git_colocated_unborn_bookmark() {
     let test_env = TestEnvironment::default();
     let workspace_root = test_env.env_root().join("repo");
-    let git_repo = git2::Repository::init(&workspace_root).unwrap();
+    let git_repo = git::init(&workspace_root);
 
+    // add a file to an (in memory) index
     let add_file_to_index = |name: &str, data: &str| {
-        std::fs::write(workspace_root.join(name), data).unwrap();
-        let mut index = git_repo.index().unwrap();
-        index.add_path(Path::new(name)).unwrap();
-        index.write().unwrap();
+        let mut tree_editor = git_repo.head_tree().map(|t| t.edit().unwrap()).unwrap_or(
+            git_repo
+                .edit_tree(gix::ObjectId::empty_tree(git_repo.object_hash()))
+                .unwrap(),
+        );
+        let blob_oid = git_repo.write_blob(data).unwrap();
+        tree_editor
+            .upsert(name, gix::object::tree::EntryKind::Blob, blob_oid)
+            .unwrap();
+        let tree_id = tree_editor.write().unwrap().detach();
+        git::checkout_tree_index(&git_repo, tree_id);
```

Comment:
You need to write the index as well.

I think we can use `git::IndexManager` after 63f8743. It should be better to not touch files other than `name`.